rbac/server/api/roles.py

```diff
@@ -63,12 +64,15 @@
     if not response:
         txn_key, txn_user_id = await utils.get_transactor_key(request)
         role_id = str(uuid4())
+        metadata1 = request.json.get("metadata")
+        metadata1 = {"sync_direction": "OUTBOUND"}
+        metadata1_str = json1.dumps(metadata1)
         batch_list = Role().batch_list(
             signer_keypair=txn_key,
             signer_user_id=txn_user_id,
             name=request.json.get("name"),
             role_id=role_id,
-            metadata=request.json.get("metadata"),
+            metadata=metadata1_str,
```

Comment:
What if the person calling this API endpoint also wants to pass in their own metadata? The way you have this coded, the only data that will be in metadata is `{"sync_direction": "OUTBOUND"}` as a string only. This is why we need a dict where we can append whatever key and values we want to the metadata field.

Reply:
No, Criss Spanton already confirmed the below statement:

**Consideration 2:** The metadata field & metadata table schema should be untouched, as per the product owner.
So, storing the new sync_direction flag is stored in the metadata field as string, as earlier.

Reply:
I understand what you're saying, but what if I pass in information into the metadata field of this API endpoint like:
`{"created_date": "04/04/2019"}`  

What your code is doing is not taking into account any user inputted information for the metadata field and you're only passing in the `metadata1_str` value.

Reply:
I totally agree to what you are suggesting. But, when I've asked the same question, I was clearly told that metadata related schema shouldn't be changed and so, 

1. I can't change the string type field to dictionary type field.
2. In the above example, if the metadata field is ingested with a dictionary data having already existing keys like created_date etc., then there needs to be some changes made at the places where you are providing that **request** with metadata. 

Anyways, if you think it as a risk, no problem. We can discuss this in the stand-up call, tomorrow.